org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/managers/MavenProjectImporterTest.java

```diff
@@ -145,6 +153,47 @@ public void testChangedProjectShouldBeUpdated() throws Exception {
 		assertEquals("Changed Project should be updated", 1, jobSpy.updateProjectJobCalled);
 	}
 
+	@Test
+	public void testPreexistingIProjectDifferentName() throws Exception {
+		File from = new File(getSourceProjectDirectory(), "maven/salut");
+		Path projectDir = Files.createTempDirectory("testImportDifferentName");
+
+		IWorkspaceRoot wsRoot = WorkspaceHelper.getWorkspaceRoot();
+		IWorkspace workspace = wsRoot.getWorkspace();
+		String projectName = projectDir.getFileName().toString();
+		IProject salut = wsRoot.getProject("salut");
+		salut.delete(true, monitor);
+		IProjectDescription description = workspace.newProjectDescription(projectName);
+		description.setLocation(new org.eclipse.core.runtime.Path(projectDir.toFile().getAbsolutePath()));
+
+		IProject project = wsRoot.getProject(projectName);
+		project.create(description, monitor);
+
+
+		assertTrue(WorkspaceHelper.getAllProjects().contains(project));
+
+		FileUtils.copyDirectory(from, projectDir.toFile());
+
+		assertTrue(project.exists());
+		Job updateJob = projectsManager.updateWorkspaceFolders(Collections.singleton(new org.eclipse.core.runtime.Path(projectDir.toString())), Collections.emptySet());
+		updateJob.join(20000, monitor);
+		assertEquals(IStatus.OK, updateJob.getResult().getSeverity());
```

Comment:
assertTrue("Failed to import testImportDifferentName:\n" + updateJob.getResult().getException(), updateJob.getResult().isOK());